src/overlays/actors/ovl_En_Bigslime/z_en_bigslime.c

```diff
@@ -54,289 +243,2911 @@ const ActorInit En_Bigslime_InitVars = {
+        (this->actionFunc == EnBigslime_FrozenFall) || (this->actionFunc == EnBigslime_Melt)) {
+        AnimatedMat_Draw(globalCtx, this->bigslimeFrozenTex);
+        gSPSegment(POLY_XLU_DISP++, 0x09, sBigslimeTargetVtx);
+        gSPDisplayList(POLY_XLU_DISP++, &D_06010B80);
+        gSPDisplayList(POLY_XLU_DISP++, &D_060105E8);
+    }
+
+    if (this->actor.scale.x > 0.0f) {
+        Matrix_SetCurrentState(&globalCtx->mf_187FC);
+        Matrix_Scale(0.0050000003f, 0.0050000003f, 0.0050000003f, MTXMODE_APPLY);
+        sysMatrix = Matrix_GetCurrentState();
+
+        for (i = 0; i < 28; i++) {
+            bubbles = &sBubbles[i];
+            dynamicVtx = &sBigslimeDynamicVtx[this->dynamicVtxState][bubbles->v];
+            sysMatrix->wx = dynamicVtx->n.ob[0] * this->actor.scale.x * bubbles->scaleVtx + this->actor.world.pos.x;
+            sysMatrix->wy = dynamicVtx->n.ob[1] * this->actor.scale.y * bubbles->scaleVtx + this->actor.world.pos.y;
+            sysMatrix->wz = dynamicVtx->n.ob[2] * this->actor.scale.z * bubbles->scaleVtx + this->actor.world.pos.z;
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, &D_0600F9D0);
+        }
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+
+    EnBigslime_SetSysMatrix(&this->actor.world.pos, globalCtx, D_06011050, this->vtxScaleX, this->vtxScaleZ,
+                            this->actor.scale.y * BIGSLIME_RADIUS_F, this->rotation, 175.0f);
+    EnBigslime_DrawGekko(&this->actor, globalCtx);
+}
+
+/* value -1: Limb Not used
+ * value 0:  GEKKO_LIMB_WAIST
+ * value 1:  GEKKO_LIMB_L_SHIN
+ * value 2:  GEKKO_LIMB_L_FOOT
+ * value 3:  GEKKO_LIMB_R_SHIN
+ * value 4:  GEKKO_LIMB_R_FOOT
+ * value 5:  GEKKO_LIMB_L_UPPER_ARM
+ * value 6:  GEKKO_LIMB_L_FOREARM
+ * value 7:  GEKKO_LIMB_L_HAND
+ * value 8:  GEKKO_LIMB_R_UPPER_ARM
+ * value 9:  GEKKO_LIMB_R_FOREARM
+ * value 10: GEKKO_LIMB_R_HAND
+ * value 11: GEKKO_LIMB_JAW
+ */
+static s8 limbPosIndex[] = {
+    -1, -1, 0, -1, 1, -1, 2, -1, 3, -1, 4, -1, 5, 6, -1, 7, 8, 9, -1, 10, -1, 11, -1, -1,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigslime/func_808ED3F4.s")
+// Some kind of offset for the position of the Gekkos right foot
+static Vec3f sRightFootOffset = { 1500.0f, 2200.0f, 0.0f };
+
+void EnBigslime_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnBigslime* this = THIS;
+    Vec3f rightFootOffset;
+
+    if (limbIndex == GEKKO_LIMB_HEAD) {
+        Matrix_GetStateTranslation(&this->actor.focus.pos);
+        this->actor.focus.rot.y = this->gekkoRot.y;
+    }
+
+    if (limbPosIndex[limbIndex] != -1) {
+        Matrix_GetStateTranslation(&this->limbPos[limbPosIndex[limbIndex]]);
+    }
+
+    if ((u32)&limbPosIndex[GEKKO_LIMB_R_ANKLE] == (u32)&limbPosIndex[limbIndex]) {
```

Comment:
Same here, are the casts necessary ?
```suggestion
    if (&limbPosIndex[GEKKO_LIMB_R_ANKLE] == &limbPosIndex[limbIndex]) {
```

Also, what an unnecessary complex comparison lol

Reply:
Oh this looks like a compiler optimization. Does it match as `limbIndex == GEKKO_LIMB_R_ANKLE`?

Reply:
It actually does! I thought I would have tried that lol. Good call